LifeCycleAssessment_Engine/Create/ElementResult.cs

```diff
@@ -0,0 +1,200 @@
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Base;
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using BH.oM.LifeCycleAssessment;
+using BH.oM.LifeCycleAssessment.MaterialFragments;
+using BH.oM.LifeCycleAssessment.Results;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+
+namespace BH.Engine.LifeCycleAssessment
+{ 
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates a list of ElementResults from a set of Identifier parameters and a list of MaterialResults. Material results are first grouped by type, and a single ElementResult is created for each type.")]
+        [InputFromProperty("objectId")]
+        [InputFromProperty("scope")]
+        [InputFromProperty("category")]
+        [Input("materialResults", "Material results used to create the element result. Results will first be grouped by type, and a single element result created per type. The phase vlaues of the element result will be the sum of the MaterialResult values.")]
+        [Output("results", "Created element results")]
+        public static List<IElementResult<MaterialResult>> ElementResults(IComparable objectId, ScopeType scope, ObjectCategory category, IEnumerable<MaterialResult> materialResults)
+        {
+            if(materialResults.IsNullOrEmpty())
+                return new List<IElementResult<MaterialResult>>();
+
+            List<IElementResult<MaterialResult>> elementResults = new List<IElementResult<MaterialResult>>();
+
+            //Group all of the provided MaterialResult by their type
+            foreach (var group in materialResults.GroupBy(x => x.GetType()))
+            {
+                //Create a single element result correpsonding to the group
+                elementResults.Add(Create.ElementResult(group.First() as dynamic, group, objectId, scope, category));
+            }
+            return elementResults;
+        }
+
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        [Description("Creates a ElementResult<T> with a type matching that of the provided MaterialResult. Element result values is computed as the sum of the values on the material results.")]
+        [Input("first", "Item of type T. Only used to help with the as dynamic casting when called from method above to help identify the actual type.")]
```

Comment:
Unsure what "type T" is. Second sentence is also unclear- only used to help with the....

Reply:
Have updated to try to explain a bit better. The reason for the input is a bit code heavy though. What I mean by that is, I needed to add the method to support the code, and workings of c#, rather than to support the logic of evaluating an element, if that makes sense. Also, this method is private, and will never be shown in any UI though, so can be a bit more relaxed potentially.

But, Have updated the desc nonetheless to try to make some better sense out of it.